### DISCUSSION TITLE
CQI Report - Subband_CQI

### DISCUSSION THREAD

**Discussion by arman-maghsoudnia** [srsran/srsRAN_Project#329] [General Help] [answered]
Dear Community,

As you may know, there are two types of CQI report in 5G standard. One of them is called "Wideband CQi", in which the UE reports the channel quality in the whole bandwidth and as a result we only have one number indicating the quality of the whole channel. Another way to report CQI is "Subband_CQI", in which the UE reports the CQI per RBG (Resource Block Group). I refer you to the following link for more information in this regard: `https://www.sharetechnote.com/html/5G/5G_CSI_Report.html#ReportingGranularity`

For my research, I need this "Subband_CQI" report. I did not find any configuration for this in the configuration list. Though, I have noticed that actually in the code there is an inline if statement that check which one (subband or wideband) is configured. The relative path to the part that I'm talking about is `lib/du_manager/converters/asn1_csi_meas_config_helpers.cpp:1006`. I dug into the code further and noticed that in `lib/scheduler/config/csi_helper.cpp:487`, the value of the type of the CQI report is set to "wideband_cqi" by default. I changed this parameter to "subband_cqi" and rebuilt the code and ran the new code. The phones connects but it fails to get the PDU session, and the CQI value printed on the console is n/A.

I wanted to ask whether this "subband_cqi| is actually implemented in the code or not. And moreover ask for other researchers' experience in this. I would appreciate if you can help me in any way to enable this "subband_cqi".

**Comment by herlesupreeth** [comment, marked as answer]
Hey @arman-maghsoudnia , Subband_CQI is not supported only wideband CQI is supported.

**Reply by arman-maghsoudnia** [reply]
Thanks for your reply @herlesupreeth. Is there any plan to implement it in the near future?

**Reply by herlesupreeth** [reply]
Yes, there is but no idea about the timeline yet.

**Reply by lmendiboure** [reply]
Hi @arman-maghsoudnia and @herlesupreeth, is this something that has now been implemented? Subband_cqi report?
